### PR TITLE
Visitor can see a list of Recipes

### PR DESCRIPTION
**Pull request by shsteindorsson**
Visitor can see a list of Recipes
========================
[PT-link](https://www.pivotaltracker.com/story/show/169667218)

## PR content
In this PR we are adding a feature where the API returns a list of all recipes.

## This PR contains:
- A unit test
- An `index` action for Recipes controller
- A Recipe serializer

### CODE REVIEW

File: spec/requests/v1/visitor_can_see_list_of_recipes_spec.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
+RSpec.describe 'GET articles index' do 
+  describe 'Listing articles' do 
+    let!(:recipes) { 2.times { create(:recipe) } }
+    let(:headers) {{ HTTP_ACCEPT: "application/json" }
+    before do
+      get '/v1/recipes'
+      headers: headers
```
Comment (tochman) [inline spec/requests/v1/visitor_can_see_list_of_recipes_spec.rb:7]: You need to make the headers part of the get request. Use a comma

File: spec/requests/v1/visitor_can_see_list_of_recipes_spec.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    end
+
+    it 'API returns 2 recipes' do
+      expect(response_json.count).to eq 2
```
Comment (tochman) [inline spec/requests/v1/visitor_can_see_list_of_recipes_spec.rb]: The question we need to consider is what the response should look like. I always advocate wrapping the collection in a key that holds the resource name. I know that you have done it differently in other projects, but I would like us to discuss this. 
```
{ "recipes" [

  ]
}
```

File: spec/requests/v1/visitor_can_see_list_of_recipes_spec.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    it 'Returns a status of 200' do
+      expect(response.status).to eq 200
+    end
+  end
+
+  describe 'Error if database contains no recipes' do 
+    before do
+      get '/v1/articles'
+      headers: headers
```
Comment (tochman) [inline spec/requests/v1/visitor_can_see_list_of_recipes_spec.rb:34]: same here. the headers are not part of the get request

File: spec/requests/v1/visitor_can_see_list_of_recipes_spec.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    end
+
+    it 'returns error status' do
+      expect(response.status).to eq 200
+    end
+
+    it 'returns error message' do
+      expect(response_json["message"]).to eq "There are no recipes here"
```
Comment (tochman) [inline spec/requests/v1/visitor_can_see_list_of_recipes_spec.rb]: This is not an error message ;-)

**Review by tochman — COMMENTED** [review]
I can't really see how the tests are supposed to pass on this one without a model and factory....

**Review by miyeaier — APPROVED** [review]

**Review by yasmineezequiel — APPROVED** [review]

File: coverage/.resultset.json [hunk 5]
```diff
@@ -0,0 +1,81 @@
+{
```
Comment (oliverochman) [inline coverage/.resultset.json]: you need to add coverage to the `.gitignore` and remove the git cache

http://www.codeblocq.com/2016/01/Untrack-files-already-added-to-git-repository-based-on-gitignore/

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: spec/requests/v1/visitor_can_see_list_of_recipes_spec.rb (hunk 1 above)
Comment (beccaburns) [reply spec/requests/v1/visitor_can_see_list_of_recipes_spec.rb:7]: change articles to recipes :)

File: spec/requests/v1/visitor_can_see_list_of_recipes_spec.rb (hunk 3 above)
Comment (beccaburns) [reply spec/requests/v1/visitor_can_see_list_of_recipes_spec.rb:34]: articles to recipes :)

**Review by beccaburns — APPROVED** [review]

File: app/controllers/v1/recipes_controller.rb [hunk 6]
```diff
@@ -1,6 +1,17 @@
...
 
 class V1::RecipesController < ApplicationController
+
+  def index
+    recipes = Recipe.all
+
+    if recipes.empty?
+      render json: { message: 'There are no recipes here' }
+    else 
+      render json: { "recipes": recipes }
```
Comment (oliverochman) [inline app/controllers/v1/recipes_controller.rb]: please serialize the response

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: app/controllers/v1/recipes_controller.rb [hunk 7]
```diff
@@ -0,0 +1,12 @@
+class V1::RecipesController < ApplicationController
+
+  def index
+    recipes = Recipe.all
+
+    if recipes.empty?
+      render_message('There are no recipes here')
```
Comment (tochman) [inline app/controllers/v1/recipes_controller.rb]: Please just render a response. W spoke about this at the meeting.

**Review by oliverochman — APPROVED** [review]

File: app/controllers/v1/recipes_controller.rb [hunk 8]
```diff
@@ -1,6 +1,17 @@
 # frozen_string_literal: true
 
 class V1::RecipesController < ApplicationController
+
+  def index
+    recipes = Recipe.all
+
+    if recipes.empty?
+      render json: { message: 'There are no recipes here' }
+    else
```
Comment (tochman) [inline app/controllers/v1/recipes_controller.rb]: I will again suggest that you just render an empty collection and deal with it on the client-side. The way you do it now will still require you to add an error handler when you receive the response.

**Review by clalil — APPROVED** [review]

**Pull request merged**